src/tests/mock/http.rs

```diff
@@ -255,3 +305,155 @@ fn full_path(path: Option<String>, query: Option<String>) -> String {
         None => path,
     }
 }
+
+/// Parses dot-separated field paths into a set of paths.
+fn parse_field_paths(paths: &[&str]) -> HashSet<Vec<String>> {
+    paths
+        .iter()
+        .map(|path| path.split('.').map(String::from).collect())
+        .collect()
+}
+
+/// Recursively compares two JSON values, excluding specified paths, and returns
+/// detailed errors using anyhow.
+fn json_matches_excluding(
+    actual: &serde_json::Value,
+    expected: &serde_json::Value,
+    exclude_paths: &HashSet<Vec<String>>,
+    current_path: &mut Vec<String>,
```

Comment:
Please can you add in the function docs what to expect to be modified in this field?

Reply:
Added docs